### ISSUE TITLE
Search Block should always have a label even if its not visible

### ISSUE THREAD

**Issue by skorasaurus** [WordPress/gutenberg#17351]
**Describe the bug**

The search block should always have a label in it even if it's not visible. 

**To reproduce**
1. Create a new page or post
2. add search block
3. delete the search label
4. publish post/page

**Expected behavior**
Even if a label is removed (we should generally discourage it, imho); there should be a default field so a user who is consuming the site wit ha screen reader or other assistive technology has a contextual clue of what the empty input field is :) 

[WordPress' Accessibility Labeling guidelines also state](https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/#labeling) that all future code should have a label as well. 

This bug was initially addressed by @afercia at https://github.com/WordPress/gutenberg/pull/16189#issuecomment-502437421

**Screenshots**

**Desktop (please complete the following information):**
all 

**Smartphone (please complete the following information):**
all

**Additional context**
This issue is still applicable even when https://github.com/WordPress/gutenberg/pull/17343 (as of https://github.com/WordPress/gutenberg/pull/17343/commits/4454bff448f7ac37eaa22d5582c015018433640c)  is applied to the code.

**Comment by afercia** [comment]
Thanks @skorasaurus for your report!

I see the accessibility feedback given on https://github.com/WordPress/gutenberg/pull/16189#issuecomment-502437421 hasn't been addressed yet and that PR was merged anyways. I'd kindly argue that's not the best way to improve collaboration between teams.

Regardless, all the points from the comment on #16189 still apply: WordPress aims to produce valid, semantic, as accessible as possible, markup.

The absence of a properly associated `<label>` element for the search field is actually a regression compared to core functions like `get_search_form()`, which makes sure there's always a label.

**Comment by skorasaurus** [comment]
I did an initial proof of concept that creates a label element of 'search' that's hidden with screen-reader-text if the label is initially removed by the user in the block editor. It's valid html but https://github.com/skorasaurus/gutenberg/commit/f5c39faed7d9b6ef314df22e0e3ad3435a8b6dff but I don't have access to a screen reader at this moment (I do on weekdays) to determine if it would work as I expect.

**Comment by skorasaurus** [comment]
I've tested this on a screen reader (JAWS 2018; chrome 72.0.3626.121) and it is working as I expected: 

What happens: 
When a user focuses onto a search input field, even when no label is present, it will say 
"search, edit, type in text". 

If you want to change the default label - I made it to be "Search" - to something else, I'm ok with it. 

I would appreciate it someone else were to test this and to make sure that my code matches the WP Guidelines.  

To test:

Create a new post 

Add a search block 

delete the label text above the search input field; 

publish post. 

(if there's an automated way to do this, that'd be super)

**Comment by epiqueras** [comment]
@skorasaurus Thanks for working on this!

Open a PR with your changes so that we can get them merged :smile:

**Comment by skorasaurus** [comment]
closed in https://github.com/WordPress/gutenberg/pull/17983 and merged \o/
